packages/graph/hash_graph/lib/graph/src/types/schema/mod.rs

```diff
@@ -1,37 +1,100 @@
+pub mod array;
+pub mod combinator;
 mod data_type;
-mod helper;
+mod object;
 pub mod property_type;
-mod reference;
 
 use core::fmt;
 
 #[doc(inline)]
 pub use self::{
+    combinator::{OneOf, ValueOrArray},
     data_type::DataType,
-    helper::{Array, OneOf, ValueOrArray},
     property_type::PropertyType,
-    reference::{DataTypeReference, PropertyTypeReference, Uri},
 };
+use crate::types::Uri;
 
 #[derive(Debug)]
 pub enum ValidationError {
-    PropertyMissing(Uri),
+    PropertyRequired(Uri),
+    PropertyMissing(usize, usize),
```

Comment:
This is a little misleading, let's rephrase to talk about MismatchedNumberOfProperties or something. There could be multiple missing properties for example.

Reply:
b1f8e04aaefec1d487f779ec5880a3ffbadd9dda